common/djangoapps/student/views.py

```diff
@@ -1885,8 +1887,32 @@ def auto_auth(request):
     create_comments_service_user(user)
 
     # Provide the user with a valid CSRF token
-    # then return a 200 response
-    if request.META.get('HTTP_ACCEPT') == 'application/json':
+    # then return a 200 response unless redirect is true
+    if redirect_when_done:
```

Comment:
The method in which this variable is read above indicates that it'll be a string and not a bool. So if it's set to anything at all, this condition will be true - which disagrees with the comment about `redirect` above.